src/utils.rs

```diff
@@ -218,6 +218,45 @@ pub fn collect_prob_dist<E: Event>(
     Ok(prob_dist)
 }
 
+pub fn filter_by_threshold<E: Event>(
+    calls: &bcf::Reader,
+    threshold: &f64,
+    out: &mut bcf::Writer,
+    events: &[E],
+    vartype: &model::VariantType
+) -> Result<(), Box<Error>> {
+    let mut record = bcf::Record::new();
+    let tags = events.iter().map(|e| e.tag_name("PROB")).collect_vec();
+    let lp_threshold = LogProb::from( PHREDProb( *threshold + 0.000000001) ); // the manual epsilon is required, because the threshold output by `control-fdr` has some digits cut off, which can lead to the threshold being lower than the values reread from the BCF record only due to a higher precision
+    loop {
+        if let Err(e) = calls.read(&mut record) {
+            if e.is_eof() {
+                return Ok(());
+            } else {
+                return Err(Box::new(e));
+            }
+        }
+
+        let variants = (utils::collect_variants(&mut record, false, false, None, false))?;
+        let mut events_probs = Vec::new();
```

Comment:
Sorry to be pedantic, but could you use `Vec::with_capacity` here?

Reply:
Sure, please be. Just learned something new about Rust. :)